src/Entity/ContactTab.php

```diff
@@ -129,33 +129,56 @@ public function setRelationships(array $relationships) {
   /**
    * {@inheritdoc}
    */
-  public function getBlock() {
-    return $this->block;
+  public function getBlock($id) {
+    if (isset($this->blocks[$id])) {
+      return $this->blocks[$id];
+    }
+    return FALSE;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function setBlock(array $block) {
+  public function getBlocks() {
+    return $this->blocks;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setBlock($id, array $block) {
     if (empty($block['id'])) {
       throw new \InvalidArgumentException('Missing required ID for block settings.');
     }
-    $this->block = $block;
+    $this->blocks[$id] = $block;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setBlocks(array $blocks) {
+    foreach ($blocks as $block) {
+      if (empty($block['id'])) {
+        throw new \InvalidArgumentException('Missing required ID for block settings.');
+      }
+    }
+    $this->blocks = $blocks;
```

Comment:
Should we make this call `$this->setBlock` or the other way round so we're not duplicating the verification logic?